### PR TITLE
Fix Fides.shouldShouldShowExperience() to return false for modal-only experiences

### PR DESCRIPTION
**Pull request by NevilleS**
Closes PROD-2763

### Description Of Changes

This adds a tiny bit of logic to the FidesJS SDK function `shouldShowExperience()` to fix a bug where we would return `true` confusingly for modal-only experiences that would not automatically show an experience to prompt for consent. The expected behaviour of this function is to return `true` only when the user should be prompted via a consent banner.

The bulk of this PR adds a thorough set of E2E tests to ensure that this SDK function is tested over all the FidesJS experience types, with & without saved consent.

### Code Changes

* [X] Fix `shouldResurfaceConsent` helper to always return false for modal-only experiences
* [X] Fix unrelated bug in `updateExperienceFromCookieConsentNotices` to correctly handle prefetched experiences without any notices (e.g. an empty experience)
* [x] Add automated E2E tests for modal, banner+modal, tcf, and empty experiences
* [x] Remove lingering references to the `show_banner` field that was removed

### Steps to Confirm

* [x] Load a default US Modal experience in Cookie House and confirm that `Fides.shouldShowExperience()` returns `false` as expected
* [x] Load a default EEA Banner + Modal experience in Cookie House and confirm that `Fides.shouldShowExperience()` returns `true` as expected and the banner is shown
* [x] Opt-in or opt-out of the EEA Banner, refresh the page, and confirm that `Fides.shouldShowExperience()` returns `false` as expected and no banner is shown
* [x] Load a default TCF Overlay experience in Cookie House and confirm that `Fides.shouldShowExperience()` returns `true` as expected and the banner is shown
* [x] Opt-in or opt-out of the TCF Overlay, refresh the page, and confirm that `Fides.shouldShowExperience()` returns `false` as expected and no banner is shown

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [x] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] if there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated
* If there are any database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!

### CODE REVIEW

File: clients/fides-js/src/lib/consent-utils.ts [hunk 1]
```diff
@@ -238,13 +238,17 @@ export const shouldResurfaceConsent = (
...
     if (experience.meta?.version_hash) {
       return experience.meta.version_hash !== cookie.tcf_version_hash;
     }
-    // Ensure we always resurface consent for TCF if for some reason experience does not have version_hash
     return true;
   }
+  // Never surface consent for modal-only experiences
+  if (experience.experience_config?.component === ComponentType.MODAL) {
+    return false;
+  }
```
Comment (NevilleS) [inline clients/fides-js/src/lib/consent-utils.ts:251]: This is the fix - it's effectively a one-line adjustment to make this utility `shouldResurfaceConsent()` behave as expected when using a modal-only experience. This method is called directly by `Fides.shouldShowExperience()` to determine whether or not the experience will be shown to the user automatically during the `init()` sequence.

(the rest of the PR is E2E tests for all the experience types)

File: clients/fides-js/src/lib/cookie.ts [hunk 2]
```diff
@@ -260,6 +260,11 @@ export const updateExperienceFromCookieConsentNotices = ({
   cookie: FidesCookie;
   debug?: boolean;
 }): PrivacyExperience => {
+  // If the given experience has no notices, return immediately and do not mutate
+  // the experience object in any way
+  if (!experience.privacy_notices) {
+    return experience;
+  }
```
Comment (NevilleS) [inline clients/fides-js/src/lib/cookie.ts:267]: This was an unrelated, but subtle bug. I found that if there was a saved consent cookie and a prefetched empty experience (e.g. `experience: {}`), this function would get called and edit the experience to become `{ privacy_notices: undefined }`, which is unexpected by the rest of the code base.

The impact of this is probably zero-to-none, it'd be a really extreme edge case where a user (1) visits a website and gets a valid experience, (2) saves their consent preference, (3) visits the website again from a different location without an experience, (4) this method causes the experience to appear non-empty but without notices, (5) client will then refetch the experience client-side, slowing down the init.

Just fixing this up.

File: clients/privacy-center/cypress/e2e/should-show-experience.cy.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    expiredTcfVersionHash?: boolean;
+    shouldShowExperience: boolean;
+  }
+
+  const testCases: TestCaseOptions[] = [
+    {
+      fixture: "experience_modal.json",
+      savedConsent: false,
+      shouldShowExperience: false,
+    },
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/should-show-experience.cy.ts:27]: This was the failing case- without the fix to `shouldResurfaceConsent`, this test fails because `shouldShowExperience` returns true unexpectedly.

File: clients/privacy-center/cypress/e2e/should-show-experience.cy.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+  ];
+
+  testCases.forEach(
+    ({
+      fixture,
+      savedConsent,
+      expiredTcfVersionHash,
+      shouldShowExperience,
+    }) => {
+      describe(`when rendering ${fixture} and saved consent ${savedConsent ? "exists" : "does not exist"} ${expiredTcfVersionHash ? "(with expired tcf_version_hash)" : ""}`, () => {
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/should-show-experience.cy.ts:78]: Thanks to the parameterized tests above, this'll exercise 9 different combinations of experiences + saved consent.

File: clients/privacy-center/cypress/e2e/should-show-experience.cy.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  interface TestCaseOptions {
+    fixture:
+      | "experience_modal.json"
+      | "experience_banner_modal.json"
+      | "experience_tcf_minimal.json"
+      | "experience_none.json";
+    savedConsent: boolean;
+    expiredTcfVersionHash?: boolean;
+    shouldShowExperience: boolean;
+  }
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/should-show-experience.cy.ts:20]: This test fixtures modify three inputs:

1. `fixture`: the type of experience to test
2. `savedConsent`: whether or not the user has a previously saved consent cookie
3. `expiredTcfVersionHash`: special case for TCF, whether the user should have a previously saved consent cookie but with an expired version hash (that doesn't match the latest experience)

It then asserts to see if `Fides.shouldShowExperience()` returns the expected result of `shouldShowExperience` in the param.

File: clients/privacy-center/cypress/e2e/show-modal.cy.ts [hunk 6]
```diff
@@ -10,9 +10,6 @@ describe("Fides.showModal", () => {
         options: {
           isOverlayEnabled: true,
         },
-        experience: {
```
Comment (NevilleS) [inline clients/privacy-center/cypress/e2e/show-modal.cy.ts:13]: This was a red herring and didn't affect these tests, removing `show_banner` entirely as it's not returned by the API anymore.

File: clients/privacy-center/cypress/fixtures/consent/experience_modal.json [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        ],
+        "language": "en",
+        "dismissable": true,
+        "show_layer1_notices": false,
+        "layer1_button_options": "opt_in_opt_out",
+        "allow_language_selection": true,
+        "auto_detect_language": true,
+        "disabled": false,
+        "id": "pri_exp-modal-000",
+        "component": "modal",
```
Comment (NevilleS) [inline clients/privacy-center/cypress/fixtures/consent/experience_modal.json:67]: Added this fixture to ensure we had a mock API response for the US-only modal component type.

File: clients/privacy-center/cypress/fixtures/consent/experience_none.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
+  "items": [],
```
Comment (NevilleS) [inline clients/privacy-center/cypress/fixtures/consent/experience_none.json:2]: Added this fixture to ensure we had a mock response for an empty experience, not matching any location.

**Review by NevilleS — COMMENTED** [review]
Couple guiding comments!
